sql/2020/22_HTTP_2/number_of_h2_and_h3_pushed_resources_and_bytes_by_content_type.sql

```diff
@@ -0,0 +1,41 @@
+# standardSQL
+# Number of H2 and H3 Pushed Resources and Bytes by Content type
+SELECT 
+  percentile,
+  client,
+  http_version,
+  content_type,
+  COUNT(DISTINCT page) AS num_pages,
+  APPROX_QUANTILES(num_requests, 1000)[OFFSET(percentile * 10)] AS pushed_requests,
+  APPROX_QUANTILES(kb_transfered, 1000)[OFFSET(percentile * 10)] AS kb_transfered,
+FROM (
+  SELECT 
+    client,
+    page,
+    JSON_EXTRACT_SCALAR(payload, '$._protocol') as http_version,
+    JSON_EXTRACT_SCALAR(payload, "$._contentType") as content_type,
+    SUM(CAST(JSON_EXTRACT_SCALAR(payload, "$._bytesIn") AS INT64)/1024) AS kb_transfered,
+    COUNT(0) AS num_requests
+  FROM 
+    `httparchive.almanac.requests`  
+  WHERE 
+    date='2020-08-01' AND
+    JSON_EXTRACT_SCALAR(payload, "$._was_pushed") = "1" AND
+    (LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "http/2" OR
+     LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "%quic%" OR
+     LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "h3%" OR
+     LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "http/3%")  
+  GROUP BY 
+    client,
+    http_version,
+    page,
+    content_type),
+  UNNEST([10, 25, 50, 75, 90]) AS percentile
+GROUP BY
+  percentile,
+  client,
+  http_version,
+  content_type
+ORDER BY
+  percentile,
+  client
```

Comment:
Suggestion to use the preprocessed `type` field rather than `$._contentType`. This groups similar types together since you don't need to differentiate between exact MIME types.

```suggestion
# standardSQL
# Number of H2 and H3 Pushed Resources and Bytes by Content type
SELECT 
  percentile,
  client,
  http_version,
  type,
  COUNT(DISTINCT page) AS num_pages,
  APPROX_QUANTILES(num_requests, 1000)[OFFSET(percentile * 10)] AS pushed_requests,
  APPROX_QUANTILES(kb_transfered, 1000)[OFFSET(percentile * 10)] AS kb_transfered,
FROM (
  SELECT 
    client,
    page,
    JSON_EXTRACT_SCALAR(payload, '$._protocol') as http_version,
    type,
    SUM(CAST(JSON_EXTRACT_SCALAR(payload, "$._bytesIn") AS INT64)/1024) AS kb_transfered,
    COUNT(0) AS num_requests
  FROM 
    `httparchive.almanac.requests`  
  WHERE 
    date='2020-08-01' AND
    JSON_EXTRACT_SCALAR(payload, "$._was_pushed") = "1" AND
    (LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "http/2" OR
     LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "%quic%" OR
     LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "h3%" OR
     LOWER(JSON_EXTRACT_SCALAR(payload, "$._protocol")) LIKE "http/3%")  
  GROUP BY 
    client,
    http_version,
    page,
    type),
  UNNEST([10, 25, 50, 75, 90]) AS percentile
GROUP BY
  percentile,
  client,
  http_version,
  type
ORDER BY
  percentile,
  client
```